src/systems/sprites.js

```diff
@@ -0,0 +1,255 @@
+    alignedPosition
+      .subVectors(vertexPosition, center)
+      .addScalar(0.5)
+      .multiply(scale);
+
+    // to check if rotation is not zero
+    if (sin !== undefined) {
+      rotatedPosition.x = cos * alignedPosition.x - sin * alignedPosition.y;
+      rotatedPosition.y = sin * alignedPosition.x + cos * alignedPosition.y;
+    } else {
+      rotatedPosition.copy(alignedPosition);
+    }
+
+    vertexPosition.copy(mvPosition);
+    vertexPosition.x += rotatedPosition.x;
+    vertexPosition.y += rotatedPosition.y;
+
+    // transform to world space
+    vertexPosition.applyMatrix4(viewWorldMatrix);
+  }
+
+  return function raycast(raycaster, intersects) {
+    worldScale.setFromMatrixScale(this.matrixWorld);
+    this.modelViewMatrix.multiplyMatrices(AFRAME.scenes[0].camera.matrixWorldInverse, this.matrixWorld);
+    viewWorldMatrix.getInverse(this.modelViewMatrix).premultiply(this.matrixWorld);
+    mvPosition.setFromMatrixPosition(this.modelViewMatrix);
+
+    //    var rotation = this.material.rotation;
+    //    var sin, cos;
+    //    if (rotation !== 0) {
+    //      cos = Math.cos(rotation);
+    //      sin = Math.sin(rotation);
+    //    }
+
+    transformVertex(vA.set(-0.5, 0.5, 0), mvPosition, CENTER, worldScale); //, sin, cos);
+    transformVertex(vB.set(0.5, 0.5, 0), mvPosition, CENTER, worldScale); //, sin, cos);
+    transformVertex(vC.set(-0.5, -0.5, 0), mvPosition, CENTER, worldScale); //, sin, cos);
+
+    uvA.set(0, 0);
+    uvB.set(1, 0);
+    uvC.set(1, 1);
+
+    // check first triangle
+    let intersect = raycaster.ray.intersectTriangle(vA, vC, vB, false, intersectPoint);
+
+    if (intersect === null) {
+      // check second triangle
+      transformVertex(vA.set(0.5, -0.5, 0), mvPosition, CENTER, worldScale); //, sin, cos);
+      uvA.set(0, 1);
+
+      intersect = raycaster.ray.intersectTriangle(vB, vC, vA, false, intersectPoint);
+      if (intersect === null) {
+        return;
+      }
+    }
+
+    const distance = raycaster.ray.origin.distanceTo(intersectPoint);
+
+    if (distance < raycaster.near || distance > raycaster.far) return;
+
+    intersects.push({
+      distance: distance,
+      point: intersectPoint.clone(),
+      uv: THREE.Triangle.getUV(intersectPoint, vA, vB, vC, uvA, uvB, uvC, new THREE.Vector2()),
+      face: null,
+      object: this
+    });
+  };
+})();
+
+export class SpriteSystem {
+  raycast(raycaster, intersects) {
+    for (let i = 0; i < this.spriteComponents.length; i++) {
+      const o = this.spriteComponents[i].el.object3D;
+      if (isVisible(o)) {
+        raycastOnSprite.call(o, raycaster, intersects);
+      }
+    }
+    return intersects;
+  }
+  constructor(scene) {
+    this.missingSprites = [];
+    this.spriteComponents = [];
+    this.maxSprites = 100;
+    Promise.all([createImageTexture(spritesheetPng), waitForDOMContentLoaded()]).then(([spritesheetTexture]) => {
+      const geometry = new THREE.BufferGeometry();
```

Comment:
I would break off these next bunch of lines into a separate function `createSpriteGeometry`.

Reply:
👍